### ISSUE TITLE
BLEND alpha mode

### ISSUE THREAD

**Issue by mittermichal** [Geodan/pg2b3dm#192]
in our project we would like to extruded geometry to be transparent to mark certain area. E.g. this red area of forest is infested with bark beetle.

For quick preview I modified MaterialCreator.MaterialBuilder to set BLEND alpha mode intead of OPAQUE:
`update ... set geom_triangle = ST_Force3D(St_SetSrid(ST_Translate(ST_Extrude(wkb_geometry,0,0,30),0,0,0.1), 4326));`

```
Height values: [0,1 m - 30,1 m]
Default color: #80FF0000
Default metallic roughness: #008000
Doublesided: True
Create glTF tiles: True
```

<img width="1145" alt="Screenshot 2024-08-12 at 12 02 15" src="https://github.com/user-attachments/assets/9397f113-cce1-4855-a70e-a362f85025c5">

I would like to make PR for this but I am not sure how you would like this to be customizable. I don't think its needed to have option to set MASK mode as whole geometry share the same base color`*`. But then should this be configurable per-geometry like `blendAlphaMode: true` in shaders json or should there be just simple switch that will enable this for all geometries or perhaps enable it per geometry anytime base color alpha channel is less than 1.
I think its better to have it per geometry because if someone has huge dataset and only wants few geometries form it to be transparent. Then it could unnecessary slow down rendering if all objects had BLEND alpha mode.

`*` https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#alpha-coverage

**Comment by bertt** [comment]
Yes it's now hardcoded to AlphaMode.OPAQUE (https://github.com/Geodan/pg2b3dm/blob/master/src/wkb2gltf.core/MaterialCreator.cs#L13)

I think the best is to add AlphaMode to the style.json (and default_alpha_mode to the options), like:

```
Default: OPAQUE
Options:    OPAQUE, MASK, BLEND

{
    "AlphaMode":[list_of_alpha_modes],
    "EmissiveColors": [list_of_emissivecolors in hex],
    "PbrMetallicRoughness": {
        "BaseColors": [ list_of_basecolors in hex],
        "MetallicRoughness": [list_of_metallic_roughness in hex]
    },
    "PbrSpecularGlossiness": {
        "DiffuseColors": [list_of_diffuse in hex],
        "SpecularGlossiness": [list_of_specular_glossiness in hex]
    }
}
```

Then you can define 1 default Alphamode for everything, 1 AlphaMode per geometry, 1 AlphaMode per inner geometry or 1 AlphaMode per triangle

**Comment by mittermichal** [comment]
thanks, will try to implement this and send PR in next 2-3 days

**Comment by mittermichal** [comment]
sure, but IMO then it should be explained to not combine with non-transparent objects
for our usecase it would be enough

**Comment by bertt** [comment]
> sure, but IMO then it should be explained to not combine with non-transparent objects for our usecase it would be enough

We can add it in the styling readme https://github.com/Geodan/pg2b3dm/blob/master/styling.md
